### PR TITLE
Updating the docs for the 0.3.0 release

### PR DESCRIPTION
**Pull request by mattfarina**
When we launch 0.3.0 I expect we can put the docs url in GH pages onto the website. This should be a little more clear for users of it there and touch on what works today.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,7 +45,7 @@ Notice that the value is a multi-line string (noted by the `|` after the `:`). H
 
 - [x] Install charts to a chart specified release name and namespace (via chart annotations)
 - [x] Install share dependencies (those that more than one application may rely on)
-- [ ] Install optional/suggests dependencies
+- [x] Install optional/suggests dependencies
 - [ ] List outdated dependencies
 - [ ] Ensure all versions resolve correctly across dependencies
```
Comment (viccuad) [inline README.md:50]: This could be argued that is already true :D, we ensure all versions are satisfied when installing.

File: docs/README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+Hypper can act as a testing ground for Helm. We do not expect that every feature
+will fit in upstream Helm.
+
+A second element in the difference is that Helm, as a package manager, does not
+handle environment deployment operations. Those are left to higher level tools
+like Helmfile (a push model like Ansible) and the Flux Helm Operator (a pull
+model like Chef). Helm expects higher level tools will use Helm and charts for
+operating in different environments.
+
+Hypper will move will make some assumptions that move it more towards the
```
Comment (viccuad) [inline docs/README.md:29]:
```suggestion
Hypper makes some assumptions that move it more towards the
```

File: docs/README.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+environment management. This is in a different way from Helmfile or the Flux
+Helm Operator but further than Helm will go.
```
Comment (viccuad) [inline docs/README.md:31]: Nit:
```suggestion
Helm Operator, but further than Helm nevertheless.
```

File: docs/README.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+### Hypper Features Not In Helm
+
+#### Installing With Release Name and Namespace Repeatedly
```
Comment (viccuad) [inline docs/README.md:35]:
```suggestion
#### Repeatable Install with defined Release Name and Namespace
```

File: docs/README.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+When dealing with system level charts you may want to have them be installed in
+the same namespace or with the same release name everywhere. For example, when
+you install a cluster wide logging service. With Hypper, you can capture this
+information as an annotation and only specify a different one if you want to
+override the default. Then when you tell Hypper to install your service it will
+be repeatedly installed to the same location.
```
Comment (viccuad) [inline docs/README.md:42]:
```suggestion
be installed to the same location, in a repeatable manner.
```

File: docs/README.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+#### Shared Dependencies
+
+Sometimes you want to have a shared system dependency. For example, you may want
+to have just a single Prometheus or Istio in a cluster. When you install an
+application that depends on this shared dependency you want to install it if not
+present and leverage the existing one if present.
+
+The additional metadata will be stored as Helm chart annotations.
```
Comment (viccuad) [inline docs/README.md:51]: Maybe expand the explanation here, for lego people. Or something like:

```suggestion
The additional metadata to signify these relationships will be stored as Helm chart annotations.
```

File: docs/design/design.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+A second element in the difference is that Helm, as a package manager, does not
+handle environment deployment operations. Those are left to higher level tools
+like Helmfile (a push model like Ansible) and the Flux Helm Operator (a pull
+model like Chef). Helm expects higher level tools will use Helm and charts for
+operating in different environments.
+
+Hypper will move will make some assumptions that move it more towards the
+environment management. This is in a different way from Helmfile or the Flux
+Helm Operator but further than Helm will go.
```
Comment (viccuad) [inline docs/design/design.md:30]: same as the other one
```suggestion
Helm Operator, but further than Helm nevertheless.
```

**Review by viccuad — COMMENTED** [review]
Looks good here, just a couple of nitpicks!

File: README.md (hunk 1 above)
Comment (mattfarina) [reply README.md:50]: It was true some time ago. We just missed updating the doc to check it off. That's all I did there.

Comment (viccuad) [reply README.md:50]: I mean, the "Ensure all versions resolve correctly across dependencies" tick.

**Review by viccuad — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (mattfarina) [reply README.md]: fixed, pushed.

**Review by viccuad — APPROVED** [review]

**Pull request merged**